system/jlib/jbuff.cpp

```diff
@@ -728,6 +728,19 @@ MemoryBuffer & MemoryBuffer::read(StringAttr & value)
     return *this;
 }
 
+MemoryBuffer & MemoryBuffer::readOpt(StringAttr & value)
+{
+    char * src = buffer + readPos;
+    size32_t len = (size32_t)strlen(src);
+    CHECKREADPOS(len+1);
+    if (len)
```

Comment:
feels like ::read(StringAttr & value) should behave this way anyway, but changing may introduce subtle bugs.

Reply:
yes.  I didn't want the possiblity of breaking existing code.  It almost certainly would.